### PR TITLE
anonymise API extension

### PR DESCRIPTION
**Pull request by joka**
API CHANGES:

- optional 'X-Anonymize' header to create/edit/delete  proposals and comments for s1 process.
  read 'docs/rest_api/anonymise.rst' and discussion in #2212 
  

BACKEND INTERNAL CHANGES:

- add ISystemUser (IUser without login/password) 		
- add anonymous  system user 			
- use anonymous user for 'user' request property if anonymize header is set
- use authenticated user for the new 'anonymized_user'  request property if anonymize header is set
- add sheet IAllowAddAnonymized to mark pool that allow  creating anoymized child resources
- add IAllowAddAnonymize sheet to process and comments service 	
- validate if anonymize header may be set 		
- set hidden anonymized creator instead of local role creator if anonymize request 

MISSING:

-  Fix broken acceptance tests (test user have different userids now
      (separated pull request)
-  optional  'X-Anonymize header to create/edit rates
       (separated pull request)
- setting to disable/enable the anonymize feature
   (can be a frontend only settingg)
-  user profile settings for defaults
    (separated pull request)
- [x] test coverage (authorization, resources.root)
- [x] make batch with 'X-Anonymized' header work

I propose we review/merge this and add the missing parts later to allow  testing the frontend implementation #2604

### CODE REVIEW

File: src/adhocracy_core/docs/anonymization.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    >>> resp = participant.post('/organisation/process', prop, extra_headers={'X-Anonymize': ''}).json
+    >>> proposal_path = resp['path']
+    >>> proposal_version_path = resp['first_version_path']
+
+
+The creator is set to anonymous::
+
+    >>> resp = participant.get(proposal_version_path)
+    >>> resp.json['data']['adhocracy_core.sheets.metadata.IMetadata']['creator']
+    '.../users/0000001/'
```
Comment (2e2a) [inline src/adhocracy_core/docs/anonymization.rst]: Maybe its better to check, if creator contains the global_anonymization_userid.

Comment (joka) [inline src/adhocracy_core/docs/anonymization.rst]: > Maybe its better to check, if creator contains the global_anonymization_userid.

yip

File: src/adhocracy_core/adhocracy_core/evolution/__init__.py [hunk 2]
```diff
@@ -888,6 +888,13 @@ def remove_participant_role_from_default_group(root,
         group_sheet.set(appstruct)
 
 
+def add_global_anonymous_user(root, registry):  # pragma: no cover
```
Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/evolution/__init__.py]: log_migration is missing

File: src/adhocracy_core/adhocracy_core/authentication/test_init.py [hunk 3]
```diff
@@ -265,3 +265,17 @@ def test_forget_return_headers_of_all_policies(
...
+    from . import AnonymizeHeader
+    request_.headers[AnonymizeHeader] = ''
+    assert is_marked_anonymize(request_) is True
+
+
+def test_is_marked_anonymize_false_if_no_anonymize_header(request_):
+    from . import is_marked_anonymize
+    from . import AnonymizeHeader
+    request_.headers[AnonymizeHeader] = ''
+    assert is_marked_anonymize(request_) is True
```
Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/authentication/test_init.py]: Check for is False without AnonymizeHeader

File: src/adhocracy_core/adhocracy_core/authentication/test_init.py [hunk 4]
```diff
@@ -279,3 +285,71 @@ def test_is_marked_anonymize_false_if_no_anonymize_header(request_):
...
+    mocker.patch('adhocracy_core.authentication.get_anonymized_creator',
+                 return_value=True)
+    assert is_created_anonymized(context)
+
+
+def test_is_created_anonymized_false_if_anonymized_creator(context, mocker):
+    from . import is_created_anonymized
+    mocker.patch('adhocracy_core.authentication.get_anonymized_creator',
+                 return_value=False)
+    assert not is_created_anonymized(context)
```
Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/authentication/test_init.py]: get_anonymized_creator should returns userid or None

Comment (joka) [inline src/adhocracy_core/adhocracy_core/authentication/test_init.py]: > get_anonymized_creator should returns userid or None
ups, to much haste

**Comment by 2e2a** [comment]
lgtm, open tasks and issues will be handles separately.

**Pull request merged**